cyraft/node.py

```diff
@@ -442,7 +454,9 @@ def _append_entries_processing(
 
         # Update current_term (if follower) (leaders will update their own term on timeout)
         if self._state == RaftState.FOLLOWER:
+            self._change_state(RaftState.FOLLOWER)  # this will reset the election timeout as well
```

Comment:
I don't think using `_change_state()` in this way is correct.

In regard to reseting the election timeout, there are three cases that we have to take into account:

1. RPC request or response contains `term T > currentTerm` (§5.1, page 4)

    -> convert to Follower (this will also reset the election timeout)

2. Regular Heartbeat (= empty AppendEntries)

    -> reset the election timeout (no `_change_state()`)

3. Regular AppendEntries

    -> rest the election timeout (no `_change_state()`)

I would suggest to use `self._reset_election_timeout()` directly.